File: ARCADE/The Core/Level 4_5: additionWithoutCarrying.cs
    using System;
     
    public class Test
    {
    	public static int additionWithoutCarrying(int param1, int param2){
            if (param1 == 0 && param2 == 0) return 0;
            int result = 0;
            string s = "";
            if (param1 < param2){
                int temp = param2;
                param2 = param1;
                param1 = temp;
            }
            while (param1 != 0){
                int a = param1 % 10;
                int b = param2 % 10;
                if (a + b >= 10) result = (a + b) % 10;
                else result = a + b;
                s = s.Insert(0, result.ToString());
                param1 = param1 / 10;
                param2 = param2 / 10;
            }
            return Convert.ToInt32(s);
        }
    	public static void Main()
    	{
    		int param1 = 456, param2 = 1734;
    		Console.Write(additionWithoutCarrying(param1, param2));
    	}
    }

File: ARCADE/The Core/Level 4_5: additionWithoutCarrying.pas
program additionWithoutCarrying;
uses Sysutils;

function additionWithoutCarrying(param1: integer; param2: integer): integer;
var temp, a, b, c : integer;
    s, s1 : string;
begin
    if((param1 = 0) and (param2 = 0)) then
    begin
        additionWithoutCarrying := 0;
        exit;
    end;
    s := ''; 
    if (param1 < param2) then
    begin
        temp := param2;
        param2 := param1;
        param1 := temp;
    end;
    
    while (param1 <> 0) do
    begin
        b := param1 mod 10;
        c := param2 mod 10;
        if (b + c >= 10) then a := (b + c)  mod 10
        else a := b + c;
        s1 := IntToStr(a);
        s := s1 + s;
        param1 := param1 div 10;
        param2 := param2 div 10;
    end;
    additionWithoutCarrying := StrToInt(s);
end;

var param1, param2 : integer;
begin
	param1 := 456; param2 := 1734;
	writeln(additionWithoutCarrying(param1, param2));
end.
